File: ios/Core/AssetPayload.swift
import Foundation

struct Thumbnail: Equatable {
  let uri: String
  let width: Int
  let height: Int
}

struct AssetPayload: Equatable {
  let uri: String
  let mime: String
  let fileName: String
  let fileSize: Int
  let width: Int
  let height: Int
  let base64: String?
  var duration: Double? = nil
  var thumbnail: Thumbnail? = nil
  var exif: ExifPayload? = nil

  var dictionary: [String: Any] {
    var dict: [String: Any] = [
      "uri": uri,
      "type": mime,
      "fileName": fileName,
      "fileSize": fileSize,
      "width": width,
      "height": height,
    ]
    if let base64 {
      dict["base64"] = base64
    }
    if let duration {
      dict["duration"] = duration
    }
    if let thumbnail {
      dict["thumbnailUri"] = thumbnail.uri
      dict["thumbnailWidth"] = thumbnail.width
      dict["thumbnailHeight"] = thumbnail.height
    }
    if let exif {
      dict["exif"] = exif.dictionary
    }
    return dict
  }
}

File: android/src/main/java/com/eugeniuszx/reactnativemediapicker/AssetPayload.kt
package com.eugeniuszx.reactnativemediapicker

internal data class Thumbnail(
  val uri: String,
  val width: Int,
  val height: Int,
)

internal data class AssetPayload(
  val uri: String,
  val mime: String,
  val fileName: String,
  val fileSize: Long,
  val width: Int,
  val height: Int,
  val base64: String?,
  val durationSeconds: Double? = null,
  val thumbnail: Thumbnail? = null,
  val exif: ExifPayload? = null,
)
